src/pudl_archiver/zenodo/api_client.py

```diff
@@ -280,8 +222,10 @@ async def add_files(self, resources: dict[str, ResourceInfo]):
             if filename not in resources and filename != "datapackage.json":
                 self.deletes.append(file_info)
 
+        self.changed = len(self.uploads or self.deletes) > 0
```

Comment:
This is the whole "reload the draft depo before trying to update the datapackage" thing.